### DISCUSSION TITLE
user is null the moment I enter the page

### DISCUSSION THREAD

**Discussion by tawjaw** [supabase/supabase#4637] [Questions]
Hello 

I am using nextjs, I am using Auth from supabase UI 

I have this to check if the user is authenticated and I am using it in every page that require authentication
```ts
export const RequireAuth = () => {
    const { user } = useUser()
    const router = useRouter()

    useEffect(() => {
        console.log("require auth", user)
        if (!user) {
            router.push('/auth')
        }
    }, [user, router])
}
```

Dashboard page 
```ts
const Dashboard: NextPage = () => {
    RequireAuth();
    const {user} = useUser();
    return (
        <SideNavApp>
        <div>
            <Button onClick={()=>{supabase.auth.signOut()}}> Logout</Button>

            <div className="bg-red-600 text-xl">HELLO {user?.email}</div>
        </div>
        </SideNavApp>

    )

}
```

when I enter the page in a new tab, the user is null and I get redirected. 
Right after that the user is not null and its authenticated. 

What am I missing? 

thanks

**Comment by mrsegev** [comment]
I have the same issue.
Before the user is initiated it returns null, and my page is redirecting to the login page.

I wish it would return `undefined` by default and then null if there is no user so we could distinguish "not ready" state from logged out.
